### ISSUE TITLE
UnidataDD2MI.xsl is not a valid version 1.0 stylesheet

### ISSUE THREAD

**Issue by aaron-sweeney** [Unidata/threddsIso#19]
Hi,

See src/main/resources/xsl/nciso/UnidataDD2MI.xsl.

From the history of this file, it appears a change was made by Rich Signell from XSL version 1.1 to 1.0 on April 18, 2015.  This causes the XSL to be invalid.  Switching back to version 1.1 seems to fix the problem.

What was the reason to revert from XSL version 1.1 to 1.0?  Can this be switched back to version 1.1?

Cordially,
Aaron

**Comment by dopplershift** [comment]
@rsignell-usgs ring a bell?

**Comment by rsignell-usgs** [comment]
I think @amilan17  made this change on April 22, 2015?https://github.com/Unidata/threddsIso/blob/a86880ea8082e2031392f425dcfb0564da8040fc/src/main/resources/xsl/nciso/UnidataDD2MI.xsl#L6

**Comment by aaron-sweeney** [comment]
@rsignell-usgs: Actually, no.  On April 22, 2015, the change you refer to was only to _document_ that the stylesheet version was changed from 1.1 to 1.0.  The change to the stylesheet _version_ from 1.1 to 1.0 was made by you on April 18, 2015.  Prior to April 18, 2015, the stylesheet was version 1.1.

Cordially,
Aaron

**Comment by rsignell-usgs** [comment]
@aaron-sweeney, okay you caught me.  ;-).   But although I submitted that change, I was only the messenger.   I went back and checked, and the change on April 18 was just me contributing a file called  UnidataDD2MI-MOD-amEdits.xsl that I received as an e-mail attachment from @amilan17 on April 16. 

I don't know anything about XSL.    So we still need @amilan17 input here.

**Comment by amilan17** [comment]
I will look at this today.


Anna
*~~~~~~Metadata Adds Meaning~~~~~~*
Anna.Milan@noaa.gov, 303-497-5099
NOAA National Centers for Environmental Information
(formerly NOAA’s National Geophysical Data Center)
http://www.ngdc.noaa.gov/metadata/emma
*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*

**Comment by amilan17** [comment]
@aaron-sweeney @rsignell-usgs 
​I found the email chain from last April about the XSLT version and emailed it to you. 

In summary, 
The original XSL was version 1.1. At that time, John Caron noticed that their Saxon HE parser was complaining about this in their TDS. I can confirm this behavior in Oxygen XML Editor. It also complains with the same configuration. ("W:[Saxon-HE 9.6.0.5] Unrecognized version 1.1: treated as 1.0")

So, at that time we changed the XSLT version to 1.0 and I started to upgrade the entire XSLT to be compliant to XSLT 2.0, but never committed that version to gitHub.

Currently, I believe that validation of the XSL varies depending on the combination of xslt version and transform engine. 

This is what I'm able to confirm in Oxygen XML Editor using the most recent gitHub XSLT. 
version 1.0 + Saxon HE, EE and PE 9.6,0.5: no problems
version 1.0 + Saxon6.5.5 and Xalan: results in fatal errors and warnings when trying to convert ncML to ISO
version 1.1 + Saxon6.5.5 and Xalan: results in fatal errors
version 1.1 + Saxon HE, EE and PE 9.6,0.5: processes with a warning

**Comment by rsignell-usgs** [comment]
@aaron-sweeney , where do you see us standing here?

**Comment by aaron-sweeney** [comment]
@rsignell-usgs I agree that validation depends on the choice of transform engine.  The reasons for the switch from XSL version 1.1 to 1.0 went undocumented.  Is it best practice to put these in the comments at the top of the relevant file, or will the existence of this issue suffice?
